pkg/server/admin.go

```diff
@@ -1494,12 +1493,13 @@ func (s *adminServer) Locations(
 	ctx context.Context, req *serverpb.LocationsRequest,
 ) (*serverpb.LocationsResponse, error) {
 	ctx = s.server.AnnotateCtx(ctx)
-
-	userName, err := userFromContext(ctx)
+	userName, isAdmin, err := s.getUserAndRole(ctx)
 	if err != nil {
 		return nil, err
 	}
-
+	if !isAdmin {
+		return nil, errInsufficientPrivilege
```

Comment:
Wondering if the better thing to do in these cases is find a good way to pass the SQL privilege error through to the UI. Otherwise we get stuck "double checking" all these instances which is both error prone and confusing.

Did you explore that option @koorosh? I think sticking to SQL as "the truth" as much as possible will be good here because the permissions model will change in the future but as long as we use the SQL-level error the UI will always be doing the right thing.

Reply:
Good question. My concern is that we have `role_members` table with an explicit `isAdmin` field to indicate users' privilege. I thought it is the source of truth right now. Will it be changed in the future?

I agree that right we don't handle permission error in a generic way with other possible SQL errors but on the other hand, restricted access for querying tables isn't the only reason for permission errors (for instance, it could be restricted depending on context).

Reply:
@dhartunian since last discussion, there was a change on server which introduced unified validation for admin permissions (2be60c1b) so now it returns "this operation requires admin privilege" error message from server.

current: 
```
This information is not available due to the current value of the 'server.remote_debugging.mode' setting.
```

message from server:
```
RequestError: this operation requires admin privilege
```
@Annebirzin , would it be okay to show server message instead of current one?

Reply:
@koorosh yep I think this is actually more clear than the current message